File: compiler/resources/immRt/FearlessError.java
package rt;

import java.nio.charset.StandardCharsets;

class FearlessError extends RuntimeException {
	public base.Info_0 info;
	public FearlessError(base.Info_0 info) {
		super();
		this.info = info;
	}
	@Override public String getMessage() { return new String(this.info.msg$imm().utf8(), StandardCharsets.UTF_8); }
	@Override public String toString() { return this.getMessage(); }
}

File: assets/rt-js/FearlessError.js
import { toStringFromUtf8 } from "./BaseStr.js";


function extractMessage(info) {
  const u8 = info.msg_m$._utf8;
  return toStringFromUtf8(u8);
}

export class FearlessError extends Error {
  constructor(info) {
    const msg = extractMessage(info);
    super(msg);                 // Initialize Error’s internal message
    this.name = "FearlessError";
    this.info = info;
  }

  getMessage() {
    return this.message;
  }

  toString() {
    return this.getMessage();
  }

  // static throwFearlessError(info) {
  //   throw new FearlessError(info);
  // }
}
